homeassistant/components/knx/__init__.py

```diff
@@ -251,16 +252,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             )
 
     hass.config_entries.async_setup_platforms(
-        entry, [platform for platform in SUPPORTED_PLATFORMS if platform in config]
+        entry,
+        [
+            platform
+            for platform in SUPPORTED_PLATFORMS
+            if platform in config and platform is not Platform.NOTIFY
+        ],
     )
 
-    # set up notify platform, no entry support for notify component yet,
-    # have to use discovery to load platform.
-    if NotifySchema.PLATFORM in conf:
+    # set up notify platform, no entry support for notify component yet
+    if NotifySchema.PLATFORM in config:
         hass.async_create_task(
-            discovery.async_load_platform(
-                hass, "notify", DOMAIN, conf[NotifySchema.PLATFORM], config
-            )
+            discovery.async_load_platform(hass, "notify", DOMAIN, {}, config)
```

Comment:
`config` here is not the original config passed to `async_setup`. We need to store the config in hass.data in `async_setup` so we can access it here and pass it correctly.

Example:
https://github.com/home-assistant/core/blob/cc3a4fef53297eb4a15d5e6cddf32bf8c3cf0928/homeassistant/components/tibber/__init__.py#L29-L33

https://github.com/home-assistant/core/blob/cc3a4fef53297eb4a15d5e6cddf32bf8c3cf0928/homeassistant/components/tibber/__init__.py#L66-L70

Reply:
Ah ok, I get the point, just some small questions about it:

We do 

```python
conf: ConfigType | None = config.get(DOMAIN)
hass.data[DATA_KNX_CONFIG] = conf
```

in `async_setup`. Is it fine we store only the integration relevant config or should we really store the complete hass config under a domain specific data string (it sounds a bit weird)? I guess the answer is no, still wanted to make sure.

Also, when we reload the integration we reload the YAML as well (this will not be reflected in the original hass config since `async_setup` is not called again during reloading, does it matter or can it just be ignored?

Additionally, when removing the integration this data key will not get deleted as we can't remove it from `hass.data` in `async_unload_entry` as that's also used for reloading the integration, how should this be handled or should it be ignored?

As per your code example I will store the complete config.

Reply:
We can ignore reloading and unloading. Notify doesn't support unloading from config entry so remaining services will probably error after unloading the config entry.

Reply:
Alright, thanks for the clarification!